docs/contribute.md

```diff
@@ -182,12 +241,60 @@ Can include a nice thumbnail.
 ![Notebook_thumbnail](https://airbus.github.io/scikit-decide/maze.png)
 ```
 
-## Unit tests
+### Adding unit tests
 
-Pytest is required to run unit tests. Providing you installed the library in developer mode as described [above](#installing-from-source-in-developer-mode), it should have been already installed by poetry.
+- Whenever adding some code, think to add some tests to the `tests/` folder.
+- Whenever fixing a bug, think to add a test that crashes before fixing the bug and does not afterwards.
 
-From the "scikit-decide" root directory, run unit tests (the "-vv" verbose mode is optional but gives additional details) with:
+Follow [above instructions](#running-unit-tests) to run them with pytest.
 
- ```shell
- poetry run pytest -vv -s tests/solvers/cpp
+## Submitting pull requests
+
+When you think you are ready to merge your modifications into the main repository, you will have to open a pull request (PR).
+We can summarize the process as follows:
+
+- Fork the repository on github.
+- Clone your fork on your computer.
+- Create a branch with a self-explanatory branch, e.g. `xx/add-solver-crazyalgoname` where "xx" stands for your initials,
+  and "crazyalgoname" for the name of the hypothetic solver you are adding.
+- Make your changes in this branch and push it to your fork.
+- Do the necessary checks (see [below](#prior-checks)).
+- Reorganize your commits (see [below](#reorganizing-commits)).
+- Submit your pull request (see [github documentation](https://help.github.com/articles/creating-a-pull-request-from-a-fork/)).
+- See if all CI checks passed on your PR
+- Wait for a review
+- Take the comments and required changes into account
+
+Note that a PR needs at least one review by a core developer to be merged.
+
+
+You may want to add a reference to the main repository to fetch from it and (re)base your changes on it:
+```shell
+git remote add upstream https://github.com/airbus/scikit-decide
 ```
+
+This [post](https://medium.com/google-developer-experts/how-to-pull-request-d75ac81449a5) points out good practices to follow to submit great pull requests and review them efficiently.
+
+
+### Prior checks
+
+Before submitting your pull request, think to
+- [run the unit tests](#running-unit-tests)
+- [check the documentation locally](#building-the-docs-locally) if you modified it
+- check you respect the coding styles by [running linters](#coding-style-and-code-linting)
+
+I you do not, you will still be able to see the status of your PR as CI will do that checks for you.
```

Comment:
2 typos: I**f** ... **these** checks for you

Reply:
done